### PR TITLE
Update README with new download features and flags

### PR DESCRIPTION
**Pull request by alimaredia**
# Changes

**Which issue is resolved by this Pull Request:**
Resolves #

**Description of your changes:**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -210,6 +210,24 @@ The full process is described graphically in the [workflow diagram](./docs/workf
 
   > **NOTE** ⏳ This command can take few minutes or immediately depending on your internet connection or model is cached. If you have issues connecting to Hugging Face, refer to the [Hugging Face discussion forum](https://discuss.huggingface.co/) for more details.
 
+  #### Downloading a specific model from a Hugging Face repository
+
+- Specify repository, model, and if necessary a hugging face token
```
Comment (nathan-weinberg) [inline README.md:215]:
```suggestion
- Specify repository, model, and a Hugging Face token if necessary
```

File: README.md [hunk 2]
```diff
@@ -210,6 +210,24 @@ The full process is described graphically in the [workflow diagram](./docs/workf
...
+
+  ```shell
+  export HF_TOKEN=<YOUR HUGGINGFACE TOKEN GOES HERE>
+  ilab download --repository=TheBloke/Mixtral-8x7B-Instruct-v0.1-GGUF --filename=mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf
+  ```
+
+  #### Downloading an entire Hugging Face repository
+
+- Specify repository, and if necessary a hugging face token. For example:
```
Comment (nathan-weinberg) [inline README.md:224]:
```suggestion
- Specify repository, and a Hugging Face token if necessary. For example:
```

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]
IIRC there's a command line option as well - worth calling that out?

File: README.md [hunk 3]
```diff
@@ -210,6 +210,24 @@ The full process is described graphically in the [workflow diagram](./docs/workf
 
   > **NOTE** ⏳ This command can take few minutes or immediately depending on your internet connection or model is cached. If you have issues connecting to Hugging Face, refer to the [Hugging Face discussion forum](https://discuss.huggingface.co/) for more details.
 
+  #### Downloading a specific model from a Hugging Face repository
+
+- Specify repository, model, and if necessary a hugging face token
+
+  ```shell
+  export HF_TOKEN=<YOUR HUGGINGFACE TOKEN GOES HERE>
```
Comment (leseb) [inline README.md:218]: If it's used once I won't recommend `export` but would simplify prefixing the `ilab` command with it.

File: README.md [hunk 4]
```diff
@@ -210,6 +210,24 @@ The full process is described graphically in the [workflow diagram](./docs/workf
...
+  ilab download --repository=TheBloke/Mixtral-8x7B-Instruct-v0.1-GGUF --filename=mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf
+  ```
+
+  #### Downloading an entire Hugging Face repository
+
+- Specify repository, and if necessary a hugging face token. For example:
+
+  ```shell
+  export HF_TOKEN=<YOUR HUGGINGFACE TOKEN GOES HERE>
```
Comment (leseb) [inline README.md:227]: ditto

**Review by hickeyma — DISMISSED** [review]
@alimaredia As discussed at yesterdays meeting, it would be great to add information on the criteria to run a mixtral model as per the examp,le.

**Review by russellb — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -210,6 +210,22 @@ The full process is described graphically in the [workflow diagram](./docs/workf
 
   > **NOTE** ⏳ This command can take few minutes or immediately depending on your internet connection or model is cached. If you have issues connecting to Hugging Face, refer to the [Hugging Face discussion forum](https://discuss.huggingface.co/) for more details.
 
+  #### Downloading a specific model from a Hugging Face repository
+
+- Specify repository, model, and a Hugging Face token if necessary
```
Comment (leseb) [inline README.md:215]: Can we point users to the HF doc about tokens? https://huggingface.co/docs/hub/en/security-tokens

File: README.md [hunk 6]
```diff
@@ -210,6 +210,22 @@ The full process is described graphically in the [workflow diagram](./docs/workf
...
+  ```shell
+  HF_TOKEN=<YOUR HUGGINGFACE TOKEN GOES HERE> ilab download --repository=TheBloke/Mixtral-8x7B-Instruct-v0.1-GGUF --filename=mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf
+  ```
+
+  #### Downloading an entire Hugging Face repository
+
+- Specify repository, and a Hugging Face token if necessary. For example:
+
+  ```shell
+  HF_TOKEN=<YOUR HUGGINGFACE TOKEN GOES HERE> ilab download --repository=mistralai/Mixtral-8x7B-v0.1
```
Comment (leseb) [inline README.md:226]: Any reason why this example is using a different repository than the one shown above?

**Review by leseb — COMMENTED** [review]
Small nit.

File: README.md (hunk 6 above)
Comment (alimaredia) [reply README.md:226]: This repository is for downloading the full model, and all of it's `.safetensors` files that need to be converted to serve or chat with. The above download command downloads the `.gguf` model that can be instantly served and chatted with by ilab. The Mistralai organization on HF doesn't have a copy of the .gguf file does the same way the other repo does.

**Review by leseb — APPROVED** [review]

**Comment by hickeyma** [comment]
@alimaredia I think you need to follow through and show how to serve the mixtral model. Otherwise, users who see the download information will have issues trying to serve the model.

**Comment by alimaredia** [comment]
@hickeyma to serve the full model it needs to be either run by a back-end that can take `.safetensors` as an input like vllm (https://docs.vllm.ai/) or the model needs to be converted to .gguf to be served by the only back-end we have implemented which is llama_cpp.

I'm working on adding additional backends (https://github.com/instructlab/instructlab/issues/1106) and will update the issue accordingly. Until support for additional backend support is upstream, are there other edits to this PR you think I should make?

**Comment by nathan-weinberg** [comment]
Going to send this through given the multiple approvals and simple level of change - we can iterate on it accordingly

**Pull request merged**